llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py

```diff
@@ -318,12 +334,15 @@ def _download_files_from_sharepoint(
 
         self._drive_id = self._get_drive_id()
 
-        self.sharepoint_folder_id = self._get_sharepoint_folder_id(
-            sharepoint_folder_path
-        )
+        if self.sharepoint_folder_id is not None:
```

Comment:
very nit: for the function signature, should be Optiona[str] for `sharepoint_folder_path` now?